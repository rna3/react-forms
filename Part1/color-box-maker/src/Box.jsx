import React from "react";

function Box({ id, width, height, color, removeBox }) {
  return (
    <div>
      <div
        style={{
          width: `${width}px`,
          height: `${height}px`,
          backgroundColor: color,
        }}
      ></div>
      <button onClick={() => removeBox(id)}>X</button>
    </div>
  );
}

export default Box;
