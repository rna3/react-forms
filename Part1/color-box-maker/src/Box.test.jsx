import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

test("renders Box without crashing", () => {
    render(<Box width={100} height={100} color="blue" />);
  });

test("matches snapshot for Box", () => {
  const { asFragment } = render(<Box width={100} height={100} color="blue" />);
  expect(asFragment()).toMatchSnapshot();
});
