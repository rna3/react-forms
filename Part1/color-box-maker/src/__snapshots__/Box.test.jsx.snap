// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for Box 1`] = `
<DocumentFragment>
  <div>
    <div
      style="width: 100px; height: 100px; background-color: blue;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
